### ISSUE TITLE
Bug with dynamic route and nested routers

### ISSUE THREAD

**Issue by AustinGil** [wesleytodd/express-openapi#20]
For some reason, I'm getting strange error while using this package with a nested router on a dynamic route. This is the error:

```
TypeError: split(...).join is not a function
    at /path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:73:47
    at Array.forEach (<anonymous>)
    at iterateStack (/path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:71:24)
    at /path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:73:7
    at Array.forEach (<anonymous>)
    at iterateStack (/path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:71:24)
    at /path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:73:7
    at Array.forEach (<anonymous>)
    at iterateStack (/path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:71:24)
    at /path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:17:5
```

The base router defines a dynamic path and requires a nested router to handle that path. It looks more or less like this:

```js
// router.js
const routes = require('express').Router();

routes.use('/:id', require('./_id'));

module.exports = routes;
```

The OpenAPI spec is used as a middleware on that nested router's base path like so:

```js
// _id.js
const router = require('express').Router({ mergeParams: true });
const { openapi } = require('../../../middleware');

router.get(
  '/',
  openapi.validPath({
    summary: 'Get a user.',
    parameters: [
      {
        in: 'path',
        imageId: 'id',
        schema: {
          type: 'integer',
        },
      },
    ],
    responses: {
      200: {
        content: {
          'application/json': {
            schema: {
              type: 'string',
            },
          },
        },
      },
    },
  }),
  async (req, res) => {
    res.send('done')
  }
);

module.exports = router;
```

If I completely remove the OpenAPI references, the application works as expected. The application also works as expected if I define the dynamic route handler in the same file as the router. It seems to only have an issue if I try using a nested router.

**Comment by wesleytodd** [comment]
Hey, sorry for the slow response (holidays) and just catching up on things.  So the error leads me to believe [this function](https://github.com/wesleytodd/express-openapi/blob/main/lib/generate-doc.js#L83-L98) is not returning an array.  Which is weird.  Could you throw a debugger or some console logs in there to see what `layer.regexp` is from the caller?

**Comment by AustinGil** [comment]
To be honest, Im not sure how best to do this. Are you saying I should add the debugger/console log in the express source code in node_modules.

**Comment by wesleytodd** [comment]
Yeah, the lines I linked above, so it would probably be in `./node_modules/@wesleytodd/openapi/lib/generate-doc.js` line 84.

If you just put something like `console.log('<<<<<<<<', thing)` there it will help because it will tell you what it got as input.  If that doesn't help, then add `console.log(split(layer.regexp))` before line 73.  We need to know what `split` is being given and returning in your setup.

**Comment by AustinGil** [comment]
Yeah, so with the setup as I have it, there is a difference between the normal routes and the dynamic route:

Normal route:
`/^\/?(?=\/|$)/i { fast_star: false, fast_slash: true } []`

Dynamic route:
`/^\/(?:([^\/]+?))\/?(?=\/|$)/i { fast_star: false, fast_slash: false } <complex:/^\/(?:([^\/]+?))\/?(?=\/|$)/i>`

When the split function is called, it returns the following which is not an array as with the other routes:
`<complex:/^\/(?:([^\/]+?))\/?(?=\/|$)/i>`

**Comment by AustinGil** [comment]
Hey @wesleytodd I put together a barebones example of what I'm working with here. Not sure if the previous comment was helpful, so maybe this will be:

https://github.com/AustinGil/express-openapi-example

**Comment by wesleytodd** [comment]
Hey, sorry for the delay again.  I have been in the process of moving.  🤷 free open source work lol

So, I confirmed this hit the fallback case which is just broken.  Here is how I did so:

1. Cloned and installed your repo
2. Added a new line with `debugger` before this line: https://github.com/wesleytodd/express-openapi/blob/main/lib/generate-doc.js#L73
3. Opened the Chrome debugger
5. Ran your server with `node --inspect --inspect-brk src/app.js`
6. ran `curl http://localhost:3001`
7. stepped into the `split` function

Figured I would share in case this helps you debug in the future.

All that said, I think this will require some debugging I am not able to embark on tonight.  Sorry if this is blocking you, but I think you might look to see if you can figure out a way around it for now from the comment where this approach was taken from: https://github.com/expressjs/express/issues/3308#issuecomment-300957572

**Comment by nklisch** [comment]
@wesleytodd I have a [PR](https://github.com/wesleytodd/express-openapi/pull/31)  up that should fix this case - it still may need some more testing, but I have some tests in there and it handles some more complex situations than before:

**Comment by pip-did** [comment]
@wesleytodd what is the status with this? The linked PR was closed but not merged, is this something still being worked on? Or are there any workarounds to make it work?

**Comment by Megapixel99** [comment]
This should be resolved by PR #56, closing issue
